sdk/src/metrics/state/temporal_metric_storage.cc

```diff
@@ -51,6 +51,36 @@ bool TemporalMetricStorage::buildMetrics(CollectorHandle *collector,
   AggregationTemporality aggregation_temporarily =
       collector->GetAggregationTemporality(instrument_descriptor_.type_);
 
+  // Fast path for single collector with delta temporality and counter, updown-counter, histogram
+  if (collectors.size() == 1 && collector->GetAggregationTemporality(
+                                    instrument_descriptor_.type_) == AggregationTemporality::kDelta)
+  {
+    // If no metrics, early return
+    if (delta_metrics->Size() == 0)
```

Comment:
Could this be moved out to be in the common code path above the check on `collectors.size() == 1`?

Reply:
Tried that, but seems the unit-tests for the path for multiple collectors is failing with the change, and I don't want to touch that path for now :) Will look into it separate of this PR.